lib/saml/ssoe_settings_service.rb

```diff
@@ -41,6 +41,12 @@ def base_settings
       def pki_needed?
         Settings.saml_ssoe.request_signing || Settings.saml_ssoe.response_encryption
       end
+
+      private
+
+      def read_file(path)
+        File.read(File.expand_path(path)) if path.present? && File.file?(File.expand_path(path))
+      end
```

Comment:
you could dry this up a bit more by setting a variable for `File.expand_path(path)`

```ruby
def read_file(path)
  return unless path.present?

  expanded_path = File.expand_path(path)

  File.read(expanded_path) if File.file?(expanded_path)
end
```

Although, I am still wondering if we should raise an error here, log it, then re-raise. That way execution stops here and we don't just silently fail when a cert doesn't exist. I believe `File.read` will raise an error if a file doesn't exist. Thoughts?

```ruby
def read_file!(path)
  raise ArgumentError, 'SSOe cert path cannot be blank' if path.blank?

  expanded_path = File.expand_path(path)

  File.read(expanded_path)
rescue => e # maybe catch specific errors here
  Rails.logger.error("Failed to read SSOe cert file: #{e.message}")
  raise
end
```